llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td

```diff
@@ -3208,6 +3208,8 @@ let Predicates = [HasSVE_or_SME] in {
   // Insert scalar into undef[0]
   def : Pat<(nxv16i8 (vector_insert (nxv16i8 (undef)), (i32 FPR32:$src), 0)),
             (INSERT_SUBREG (nxv16i8 (IMPLICIT_DEF)), FPR32:$src, ssub)>;
+  def : Pat<(nxv16i8 (vector_insert (nxv16i8 (undef)), (i64 FPR64:$src), 0)),
```

Comment:
We don't really want this because it goes against the canonical form where existing code expects inserts into vectors whose element type is `i32` or smaller to take an `i32` based scalar.